### DISCUSSION TITLE
static ip while flashing device with ESPixelStick Flash Tool

### DISCUSSION THREAD

**Discussion by marcobrianza** [forkineye/ESPixelStick#662] [Q&A]
I have tried disabling "Use DHCP" and setting a static ip in:

FW Version 4.0-ci5943922254
Build Date Aug 22 2023 - 20:53:21
Architecture ESP32
Board Name d32_pro

the setting is not applied and DHCP is enabled

**Comment by MartinMueller2003** [comment]
please start a new session, configure your settings, perform an upload to the device, press the save logs button. Attach a screenshot of the flashtool with its configuration and the logs to this conversation. I cant really help without that.

**Reply by marcobrianza** [reply]
![screen](https://github.com/forkineye/ESPixelStick/assets/1008278/b2b5675b-db3c-4606-a35f-855886628112)
[logs.txt](https://github.com/forkineye/ESPixelStick/files/12591050/logs.txt)

here they are, thank you

**Comment by MartinMueller2003** [comment]
Those all look exactly like they should. There is another button for downloading your configuration to the flashtool. This will create a download file in the dist directory. In a subdirectory you will find the config.json. Please attach it to the conversation.

**Comment by marcobrianza** [comment]
the download configuration fails with this errors
![e1](https://github.com/forkineye/ESPixelStick/assets/1008278/aafcca67-7b47-463f-9b08-5c63da7cd381)
![e2](https://github.com/forkineye/ESPixelStick/assets/1008278/7d32f8d1-7f2d-4db8-9a9c-37b7aaee0320)
[log2.txt](https://github.com/forkineye/ESPixelStick/files/12593240/log2.txt)

anyway picking the config.json from the fs folder i get this

{
  "cfgver": 1.0,
  "init": true,
  "device": {
    "id": "d32-pro-id-aa"
  },
  "network": {
    "ssid": "colors",
    "passphrase": "colors01",
    "hostname": "d32-pro-aa",
    "ip": "192.168.8.33",
    "netmask": "255.255.255.0",
    "gateway": "192.168.8.254",
    "dhcp": true,
    "sta_timeout": 15.0,
    "ap_fallback": true,
    "ap_timeout": 120.0,
    "ap_reboot": true
  }
}

It might just be a problem of the flashing tool that does not set the "dhcp" to false

**Comment by MartinMueller2003** [comment]
That is what I suspect as well. That is why I wanted the config file from the ESP. But if you have DHCP set too false, and that file shows true then there is a problem in the flashtool

**Reply by marcobrianza** [reply]
who is maintaining the tool?

**Comment by MartinMueller2003** [comment]
Shelby is primary and I do some work on it as needed.

**Reply by marcobrianza** [reply]
do I need to open an issue for it, or you can manage it?

**Comment by MartinMueller2003** [comment]
This has been fixed and a new tool is used to flash: https://espixelstickwebflasher.from-ct.com:5000/
